pkg/util/parsing.go

```diff
@@ -49,10 +35,42 @@ func ParseOpaquePorts(override string, containers []corev1.Container) []string {
 	return values
 }
 
-// ToPortRanges converts a slice of strings into a slice of PortRanges.
-func ToPortRanges(portRanges []string) []*config.PortRange {
-	ports := make([]*config.PortRange, len(portRanges))
-	for i, p := range portRanges {
+// isNamed checks if a port range is actually a container named port (e.g.
+// `123-456` is a valid name, but also is a valid range); all port names must
+// be checked before making it a list.
+func isNamed(pr string, containers []corev1.Container) (int32, bool) {
+	for _, c := range containers {
+		for _, p := range c.Ports {
+			if p.Name == pr {
+				return p.ContainerPort, true
+			}
+		}
+	}
+	return 0, false
+}
+
+// ParseOpaquePorts parses the opaque ports annotation into a list of ports
+func ParseOpaquePorts(override string) []string {
```

Comment:
Is this specific to Opaque Ports? Or is this really just `ParsePortRanges`

Also, from docs, it's unclear how this differs from `GetPortRanges` and why it returns a list of strings. I would expect parsing ports to produce a list of port numbers...

Reply:
We don't actually use this anymore so I can remove this.

Reply:
It's purpose got split up into `ParseContainerOpaquePorts` and `parseServiceOpaquePorts`.